.github/workflows/lint-with-ruff.yml

```diff
@@ -0,0 +1,41 @@
+---
+name: Ruff Lint
+
+on:
+  workflow_call:
+
+jobs:
+  ruff-lint:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: read
+    steps:
+      - name: Checkout head
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+
+      - name: Setup python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11' 

```

Comment:
Is not this excessive? Python 3.10 included in `ubuntu-latest` seems to be enough.

Reply:
VyOS 1.5 uses Python 3.11 so that's a good match using 3.11 here

Reply:
Ok, i will keep as 3.11

Reply:
@c-po, sorry, I was not clear enough.

The workflow seems to have no Python dependencies except this single file: https://github.com/ChartBoost/ruff-action/blob/main/action/main.py, and I do not see any reason to update Python.

Reply:
@zdc My assumption is that python 3.11 we are setting up here to get the linting/issues for scanned files as per Python 3.11. 
I remember the lint results used to differ as per python version used. I can verify that

Reply:
> @zdc My assumption is that python 3.11 we are setting up here to get the linting/issues for scanned files as per Python 3.11. I remember the lint results used to differ as per python version used. I can verify that

Not in this case. Ruff is completely portable and does not need any dependencies, except standard C libraries.

Reply:
> > @zdc My assumption is that python 3.11 we are setting up here to get the linting/issues for scanned files as per Python 3.11. I remember the lint results used to differ as per python version used. I can verify that
> 
> Not in this case. Ruff is completely portable and does not need any dependencies, except standard C libraries.

Ok @zdc Removed py version install step